lib/Raven/Client.php

```diff
@@ -559,9 +559,22 @@ protected function get_http_data()
         if (!empty($_POST)) {
             $result['data'] = $_POST;
         }
+        
+        // if the $_POST super global is empty, try to retrieve a json string
+        // prior to PHP 5.6, a stream opened with php://input could only be read once, so only do this with 5.6 or up
+        if (version_compare(PHP_VERSION, '5.6', '>=') && empty($_POST)) {
+            try {
+                $data = file_get_contents('php://input');
+                $result['data'] = json_decode($data, true);
```

Comment:
The point is to TRY to parse it if it happens to be Json data. I suppose if it fails to parse (without an exception) I should not set 'data'.